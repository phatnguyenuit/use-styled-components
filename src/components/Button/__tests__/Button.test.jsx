import { Button, ReversedButton, TomatoButton } from "../index";

import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("Test Buttons with `react-test-renderer`", () => {
  test("should render `Button` without error", () => {
    const button = renderer.create(<Button>Button</Button>);
    let tree = button.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("should render `TomatoButton` without error", () => {
    const button = renderer.create(<TomatoButton>Button</TomatoButton>);
    let tree = button.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("should render `ReversedButton` without error", () => {
    const button = renderer.create(<ReversedButton>Button</ReversedButton>);
    let tree = button.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Test Buttons with `enzyme`", () => {
  test("`Button` should render with text = `Button`", () => {
    const button = shallow(<Button>Button</Button>);
    expect(button.contains("Button")).toEqual(true);
  });
  test("`Button` should render with primary = true", () => {
    const button = shallow(<Button primary>Button</Button>);
    expect(button.props("primary")).toBeTruthy();
  });
  test("`TomatoButton` should render with text = `TomatoButton`", () => {
    const button = shallow(<TomatoButton>TomatoButton</TomatoButton>);
    expect(button.contains("TomatoButton")).toEqual(true);
  });
  test("`ReversedButton` should render with reversed text of `ReversedButton`", () => {
    const text = "ReversedButton";
    const button = shallow(<ReversedButton>{text}</ReversedButton>);
    const reversedText = text.split("").reverse();
    expect(button.contains(reversedText)).toEqual(true);
  });
});
