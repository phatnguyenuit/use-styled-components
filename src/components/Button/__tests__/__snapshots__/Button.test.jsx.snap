// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Buttons with \`react-test-renderer\` should render \`Button\` without error 1`] = `
<button
  className="sc-bdVaJa iQJFUl"
>
  Button
</button>
`;

exports[`Test Buttons with \`react-test-renderer\` should render \`ReversedButton\` without error 1`] = `
<button>
  n
  o
  t
  t
  u
  B
</button>
`;

exports[`Test Buttons with \`react-test-renderer\` should render \`TomatoButton\` without error 1`] = `
<button
  className="sc-bdVaJa sc-bwzfXH jYytoe"
>
  Button
</button>
`;
